src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1664,12 +1655,12 @@ private void emitPushClauseArray(int clauseDataSlot, int which) {
 
     private void emitZero(BasicType type) {
         switch (type) {
-            case I_TYPE: mv.visitInsn(Opcodes.ICONST_0); break;
-            case J_TYPE: mv.visitInsn(Opcodes.LCONST_0); break;
-            case F_TYPE: mv.visitInsn(Opcodes.FCONST_0); break;
-            case D_TYPE: mv.visitInsn(Opcodes.DCONST_0); break;
-            case L_TYPE: mv.visitInsn(Opcodes.ACONST_NULL); break;
-            default: throw new InternalError("unknown type: " + type);
+            case I_TYPE -> mv.visitInsn(Opcodes.ICONST_0);
+            case J_TYPE -> mv.visitInsn(Opcodes.LCONST_0);
+            case F_TYPE -> mv.visitInsn(Opcodes.FCONST_0);
+            case D_TYPE -> mv.visitInsn(Opcodes.DCONST_0);
+            case L_TYPE -> mv.visitInsn(Opcodes.ACONST_NULL);
+            default -> throw new InternalError("unknown type: " + type);
```

Comment:
perhaps
```
  mv.visitInsn(switch(type) { ...
```

Reply:
Hi Remi, thanks for the suggestion. I've added that in now. See commit a8706b0